### PR TITLE
HOCS-2136: Transfer to OGD at MPAM Creation stage

### PR DESCRIPTION
**Pull request by parth-ho**
This PR contains the functionality to input MPAM missalocations data the data input stage.

### CODE REVIEW

File: server/middleware/validation.js [hunk 1]
```diff
@@ -139,24 +159,39 @@ function validationMiddleware(req, res, next) {
...
     next();
 
-    function isFieldVisible(visibilityConditions, data) {
+    function isFieldVisible(visibilityConditions, hideConditions, data) {
+        let i;
         let isVisible = true;
         if (visibilityConditions) {
             isVisible = false;
-            for (var i = 0; i < visibilityConditions.length; i++) {
+            for (i = 0; i < visibilityConditions.length; i++) {
```
Comment (zuzak) [inline server/middleware/validation.js:167]: I'd put `let i` here (I know if it were `var` JavaScript hoisting would mean it made no difference, but I think here it'd mean the variable disappears when we exit the `for` loop, which would be safer). See `isComponentVisible()` further down this PR.

File: src/shared/common/forms/form-repository.jsx [hunk 2]
```diff
@@ -49,15 +49,28 @@ function renderFormComponent(Component, options) {
...
             if (data[condition.conditionPropertyName] && data[condition.conditionPropertyName] === condition.conditionPropertyValue) {
                 isVisible = true;
             }
         }
     }
+
+    // hide component based on hideConditions
+    if (hideConditions) {
+        for (let i = 0; i < hideConditions.length; i++) {
+            const condition = hideConditions[i];
```
Comment (zuzak) [inline src/shared/common/forms/form-repository.jsx:68]: Does JS mind multiple `const` here?

File: server/middleware/validation.js [hunk 3]
```diff
@@ -139,24 +159,39 @@ function validationMiddleware(req, res, next) {
...
                 const condition = visibilityConditions[i];
                 if (data[condition.conditionPropertyName] && data[condition.conditionPropertyName] === condition.conditionPropertyValue) {
                     isVisible = true;
                 }
             }
         }
+
+        if (hideConditions) {
+            isVisible = true;
+            for (i = 0; i < hideConditions.length; i++) {
```
Comment (zuzak) [inline server/middleware/validation.js:177]: Consider:
```js
for (let condition of hideConditions) {
```

**Review by zuzak — COMMENTED** [review]
You've added a merge commit to your pull request. Ideally you'd rebase instead, so we don't have references in `master` to branches we shortly don't care about.

You could rebase it away using `git rebase -i origin/master` and changing the word next to that commit to `drop`.

File: src/shared/common/forms/form-repository.jsx (hunk 2 above)
Comment (parth-ho) [reply src/shared/common/forms/form-repository.jsx:68]: Seem this is fine: https://stackoverflow.com/questions/50807989/es6-javascript-const-inside-or-let-outside-loop

**Pull request merged**